src/tls.c

```diff
@@ -109,6 +109,34 @@ static int ssl_seed(void)
   return 0;
 }
 
+void verify_cert_expiry(int idx) {
+  X509 *x509;
+
+#if OPENSSL_VERSION_NUMBER >= 0x10002000L /* 1.0.2 */
+  x509 = SSL_CTX_get0_certificate(ssl_ctx);
+#else
+  BIO *bio = BIO_new_file(tls_certfile, "r");
+  if (!bio)
+    return;
+  x509 = PEM_read_bio_X509(bio, NULL, NULL, NULL);
+#endif
+  if (x509) {
+    if (X509_cmp_current_time(X509_get_notAfter(x509)) < 0) {
+      if (idx) {
+        dprintf(idx, "WARNING: SSL/TLS certificate %s expired\n", tls_certfile);
+        dprintf(idx, "You can generate new certificates by running 'make sslcert' from the source directory\n\n");
+      } else {
+        putlog(LOG_MISC, "*", "\nWARNING: SSL/TLS certificate %s expired", tls_certfile);
+        putlog(LOG_MISC, "*", "You can generate new certificates by running 'make sslcert' from the source directory\n");
+      }
+    }
+#if OPENSSL_VERSION_NUMBER < 0x10002000L /* 1.0.2 */
+  X509_free(x509);
```

Comment:
why is the cleanup unnecessary in higher openssl versions?

Reply:
1. x509 see https://man.openbsd.org/OpenBSD-7.6/SSL_CTX_get0_certificate.3:

The returned pointer must not be freed by the caller.

2. bio is only declared for OPENSSL_VERSION_NUMBER < 0x10002000L:

https://github.com/eggheads/eggdrop/pull/1411/files#diff-bb132d62ad3f3a5eac6006e2a383ad909e30072360265a49a962ac765962be34R115-R118

Reply:
while at it i noticed i had to fix the indent. done.